src/components/FeatureViews/Registration/RegistrationCourseEnrollments.js

```diff
@@ -91,24 +98,48 @@ const RegistrationCourseEnrollments = ({courseID}) => {
             }), {}));
     }, [course.roster]);
 
+    useEffect(()=>{
+        api.fetchAllSessions({
+            "config": {
+                "params": {
+                    "time_frame": "month",
+                    "time_shift": 0,
+                },
+            },
+            "id": "",
+        });
+    },[api]);
+
     const loadedStudents = useMemo(() =>
         course.roster.filter((studentID) => students[studentID])
     , [course.roster, students]);
 
+    const currentMonthSessions = sessionArray(sessions);
+
     // no students enrolled
     if (course.roster.length === 0) {
         return <NoListAlert list={"Enrolled Students"}/>;
     }
 
-    if (loadedStudents.length === 0) {
-        if (hooks.isLoading(studentStatus)) {
-            return <Loading />;
+    if (loadedStudents.length === 0 || !currentMonthSessions) {
+        if (hooks.isLoading(studentStatus) || !currentMonthSessions) {
+            return <Loading small />;
         }
         if (hooks.isFail(studentStatus)) {
             return "Error loading enrollment details!";
         }
     }
 
+    const today = new Date();
+    const upcomingSession = currentMonthSessions
+        .filter((session) => ((Number(session.course) === Number(courseID)) &&
+            new Date(session.start_datetime) >= today))
+        .sort((sessionA, sessionB) => {
+            const distanceA = Math.abs(today - sessionA);
+            const distanceB = Math.abs(today - sessionB);
+            return distanceA - distanceB;
+        })[0];
```

Comment:
Is this just trying to find the earliest date? You can simple sort by `sessionA - sessionB` if so.